### PR TITLE
Moving responsiblity of Lambda role creation to application itself

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: ChildAccountRole.template.yaml [hunk 1]
```diff
@@ -55,10 +55,7 @@ Resources:
             - iam:AttachRolePolicy
             - iam:PutRolePolicy
             - iam:DeleteRolePolicy
-            Resource:
-            - !Sub arn:aws:iam::${AWS::AccountId}:role/SampleLambdaRole-*
-            - !Sub arn:aws:iam::${AWS::AccountId}:role/CodePipelineDeploymentRole-*
-            - !Sub arn:aws:iam::${AWS::AccountId}:role/CodePipelineServiceRole-*
+            Resource: !Sub arn:aws:iam::${AWS::AccountId}:role/*
```
Comment (janbj) [inline ChildAccountRole.template.yaml:59]: For use with future Lambda roles

File: sample-project/README.md [hunk 2]
```diff
@@ -3,4 +3,4 @@ $(aws cloudformation list-exports --query "Exports[?Name=='SamBucket'].Value" --
 
 
 aws cloudformation package --template-file sample.yaml --output-template-file sample-transformed.yaml --s3-bucket $(aws cloudformation list-exports --query "Exports[?Name=='SamBucket'].Value" --output text)
-aws cloudformation deploy --template-file sample-transformed.yaml --stack-name sample-project  --capabilities CAPABILITY_IAM
\ No newline at end of file
+aws cloudformation deploy --template-file transformed.yaml --stack-name sample-project  --capabilities CAPABILITY_IAM
```
Comment (janbj) [inline sample-project/README.md:6]: One of many changes to move away from the sample.

File: sample-project/buildspec.build.yaml [hunk 3]
```diff
@@ -17,12 +17,12 @@ phases:
       - echo "The package command will put the Lambda function code into S3"
       - echo "It will then create a new version of the template file with the S3 url to the source code"
       - echo "Learn more at https://docs.aws.amazon.com/lambda/latest/dg/serverless_app.html"
-      - aws cloudformation package --template-file sample.yaml --output-template-file sample-transformed.yaml --s3-bucket $SAM_BUCKET
+      - aws cloudformation package --template-file sample.yaml --output-template-file transformed.yaml --s3-bucket $SAM_BUCKET
       - |-
-        echo "{ \"Parameters\" : { \"MasterSecretArn\" : \"$SECRET_ARN_DEV\", \"MasterSecretsManagerKey\" : \"$SECRET_MANAGER_KEY\", \"Stage\" : \"dev\" } }" > sample-configuration-dev.json
-        echo "{ \"Parameters\" : { \"MasterSecretArn\" : \"$SECRET_ARN_PROD\", \"MasterSecretsManagerKey\" : \"$SECRET_MANAGER_KEY\", \"Stage\" : \"prod\" } }" > sample-configuration-prod.json
+        echo "{ \"Parameters\" : { \"MasterSecretsManagerKey\" : \"$SECRET_MANAGER_KEY\", \"Stage\" : \"dev\" } }" > configuration-dev.json
```
Comment (janbj) [inline sample-project/buildspec.build.yaml:22]: We don't need a master secret for our application at this point.

File: sample-project/sample.yaml [hunk 4]
```diff
@@ -13,6 +14,20 @@ Parameters:
     Type: String
 
 Resources:
+  SampleLambdaRole:
```
Comment (janbj) [inline sample-project/sample.yaml:17]: Creating the Lambda role inside application now.

File: templates/project.template.yaml [hunk 5]
```diff
@@ -49,79 +49,10 @@ Conditions:
   AddProdAccountPermissions: !Not [!Equals [ !Ref ProdAwsAccountId, '' ]]
 
 Resources:
-  SecretDev:
```
Comment (janbj) [inline templates/project.template.yaml:52]: We don't need this for our applications currently.

File: templates/project.template.yaml [hunk 6]
```diff
@@ -140,11 +71,11 @@ Resources:
               AWS:
                 - !If
                   - AddDevAccountPermissions
-                  - !Sub arn:aws:iam::${DevAwsAccountId}:role/SampleLambdaRole-${AWS::Region}-${DevAwsAccountId}-dev
+                  - !Sub ${DevAwsAccountId}
```
Comment (janbj) [inline templates/project.template.yaml:76]: Extending access to all roles and identities inside account to be able to use from future Lambda roles.

**Review by brinxmat — APPROVED** [review]

**Review by torarnet — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**